File: LeetCode/Go/590-tree-n叉树的后序遍历.go
/*
 * @lc app=leetcode.cn id=590 lang=golang
 *
 * [590] N叉树的后序遍历
 */
package leetcode

// @lc code=start
func postorder(root *Node) []int {
	res := []int{}
	
	var post func(*Node)
	post = func(root *Node){
		if root == nil {
			return
		}
		for _, child := range root.Children {
			post(child)
		}
		res = append(res, root.Val)
	}

	post(root)
	return res
}
// @lc code=end

File: LeetCode/C/590-tree-n-叉树的后序遍历.cpp
/*
 * @lc app=leetcode.cn id=590 lang=cpp
 *
 * [590] N 叉树的后序遍历
 */
#include "main.h"
// @lc code=start
class Solution {
  public:
    vector<int> postorder(Node *root) {
        vector<int> res;
        post(root, res);
        return res;
    }

    void post(Node *root, vector<int> &res) {
        if (root == nullptr)
            return;
        for (auto child : root->children)
            post(child, res);
        res.push_back(root->val);
    }
};
// @lc code=end
